Dangerfile

```diff
@@ -296,6 +297,22 @@ def validateSwiftFiles(file)
   end
 end
 
+def checkHardcodedNib(file)
+  File.foreach(file) do |line|
+    line = line.gsub('\n','').strip
+    ext = File.extname(file)
+    case ext
+    when ".xib"
+      if line =~ /<color(?!.*(?:white="1"|white="0\.0"|red="0\.93725490196078431" green="0\.93725490196078431" blue="0\.95686274509803926" alpha="1"|red="0\.93725490199999995" green="0\.93725490199999995" blue="0\.95686274510000002" alpha="1"|red="0\.0" green="0\.0" blue="0\.0"|red="1" green="1" blue="1" alpha="1"|key="textColor" cocoaTouchSystemColor="darkTextColor"\/>|key="titleShadowColor"|cocoaTouchSystemColor|<nil|<view|userDefinedRuntimeAttribute|<string key="text"|image|<\/string>|<label|<document)).*\/>/
```

Comment:
@ghashi @eiditakahashi 
Any way we can break this into smaller declarations? This regex as it is now would be nearly impossible to debug and maintain.

At the very least, [breaking the regex into multiple lines](http://stackoverflow.com/questions/3762183/split-ruby-regex-over-multiple-lines) would really help the next person to understand what's being done.

I was thinking something along the lines of:
```ruby
COLOR_WHITE_1                  = white="1"
COLOR_WHITE_0                  = white="0\.0"
COLOR_GRAY_ARBITRARY_1         = red="0\.93725490196078431" green="0\.93725490196078431" blue="0\.95686274509803926" alpha="1"
COLOR_GRAY_ARBITRARY_2         = red="0\.93725490199999995" green="0\.93725490199999995" blue="0\.95686274510000002" alpha="1"
COLOR_BLACK                    = red="0\.0" green="0\.0" blue="0\.0"
COLOR_WHITE_2                  = red="1" green="1" blue="1" alpha="1"
ATTR_TEXT_COLOR                = key="textColor" cocoaTouchSystemColor="darkTextColor"\/>
ATTR_TITLE_SHADOW_COLOR        = key="titleShadowColor"
COCOA_TOUCH_SYSTEM_COLOR       = cocoaTouchSystemColor
TAG_NIL                        = <nil
TAG VIEW                       = <view
USER_DEFINED_RUNTIME_ATTRIBUTE = userDefinedRuntimeAttribute
TAG_STRING                     = <string key="text"
IMAGE                          = image
TAG_STRING_EMPTY               = <\/string>
TAG_LABEL                      = <label
TAG_DOCUMENT                   = <document


if line =~ /
<color(
  ?!.*(
    ?:#{Regexp.quote(COLOR_WHITE_1)}
    | #{Regexp.quote(COLOR_WHITE_0)}
    | #{Regexp.quote(COLOR_GRAY_ARBITRARY_1)}
    | #{Regexp.quote(COLOR_GRAY_ARBITRARY_2)}
    | #{Regexp.quote(COLOR_BLACK)}
    | #{Regexp.quote(COLOR_WHITE_2)}
    | #{Regexp.quote(ATTR_TEXT_COLOR)}
    | #{Regexp.quote(ATTR_TITLE_SHADOW_COLOR)}
    | #{Regexp.quote(COCOA_TOUCH_SYSTEM_COLOR)}
    | #{Regexp.quote(TAG_NIL)}
    | #{Regexp.quote(TAG)}
    | #{Regexp.quote(USER_DEFINED_RUNTIME_ATTRIBUTE)}
    | #{Regexp.quote(TAG_STRING)}
    | #{Regexp.quote(IMAGE)}
    | #{Regexp.quote(TAG_STRING_EMPTY)}
    | #{Regexp.quote(TAG_LABEL)}
    | #{Regexp.quote(TAG_DOCUMENT)}
  )
)
.*\/>/
```